src/main/java/duke/AddDeadlineCommand.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.FormatStyle;
+
+/**
+ * Encapsulates a Add Deadline Command.
+ */
+public class AddDeadlineCommand extends Command {
+    /**
+     * Instantiates a AddDeadlineCommand.
+     * @param parsedCommand the parsed command
+     */
+    public AddDeadlineCommand(String[] parsedCommand) {
+        super(parsedCommand);
+    }
+
+    /**
+     * Executes command and write to storage.
+     * @param tasks the tasklist containing tasks so far
+     * @param ui ui to interact with user
+     * @param storage storage to read and write to storage file
+     * @throws DukeException if parsedCommand does not meet the requirements
+     */
+    @Override
+    void execute(TaskList<Task> tasks, Ui ui, Storage storage) throws DukeException {
+        // if length is not 2, nothing was passed in after 'deadline'
+        if (getParsedCommand().length != 2) {
+            throw new DukeException("NANI??! Enter a description for your deadline!\n");
+        }
+
+        // if description is lacking a /by keyword
+        String description = getParsedCommand()[1];
+        if (description.indexOf("/by") < 0) {
+            throw new DukeException("Please enter a valid deadline! Remember to add '/by'\n");
+        }
+
+        String[] descriptionArray = description.split("/by");
+        if (descriptionArray.length != 2) {
+            throw new DukeException("NANI??! Enter your deadline name, date & time(if any) (e.g. 2020-01-30 18:30)!\n");
+        }
+
+        String deadlineName = descriptionArray[0];
+        String deadlineEndTime = descriptionArray[1];
+
+        // split to date and time
+        String[] deadlineEndingArray = deadlineEndTime.trim().split(" ");
+        // create a LocalDate object
+        String formattedDeadline;
+        try {
+            LocalDate lt = LocalDate.parse(deadlineEndingArray[0]);
+            formattedDeadline = lt.format(DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG));
+            if (deadlineEndingArray.length == 2) {
+                LocalTime t = LocalTime.parse(deadlineEndingArray[1]);
+                formattedDeadline += " "
+                        + DateTimeFormatter.ofPattern("hh:mm a").format(t);
+            }
+        } catch (DateTimeParseException e) {
+            throw new DukeException("NANI??! Enter a valid date and time(optional).\n"
+                    + "Date should be: yyyy-mm-dd.\n"
+                    + "Time should be HH:mm");
+        }
+
+        Task taskToAdd = new Deadline(deadlineName, formattedDeadline);
+        addTask(tasks, taskToAdd, ui);
+    }
+
+    /**
+     * Adds a task to the tasklist.
+     * @param tasks the tasklist containing tasks so far
```

Comment:
For the javadocs, there should be a spacing between description and @param

Reply:
Wow great catch! You are most definitely right.